src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,61 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename,
+                             std::ios_base::iostate flag,
+                             std::ios_base::openmode mode)
+    : filename_(filename), flag_(flag), mode_(mode)
+{
+  if (fs::exists(filename_)) {
```

Comment:
Move the deletion until the very last rename() in the destructor so that the original file is not deleted upon a failed write.